File: src/tests/integration/Bc.Endpoint.Integration.Tests/CommentAnswerNotificationSender.cs
using System;
using System.Threading.Tasks;
using Bc.Contracts.Internals.Endpoint.CommentAnswerNotification.Commands;
using NServiceBus;
using NUnit.Framework;

namespace Bc.Endpoint.Integration.Tests
{
    [TestFixture]
    [Ignore("Only for manual tests")]
    public class CommentAnswerNotificationSender
    {
        private IEndpointInstance endpointInstance;
        private readonly Guid commentId = Guid.Parse("9a3a0fdc-3a5a-4862-845c-09792d2b6f9a");

        [SetUp]
        public async Task SetUp()
        {
            this.endpointInstance = await EndpointFactory.GetSenderEndpoint().ConfigureAwait(false);
        }

        [TearDown]
        public Task TearDown()
        {
            return this.endpointInstance.Stop();
        }

        [Test]
        public async Task RegisterCommentNotification_Send_NoException()
        {
            // Arrange
            const string userEmail = "test@test.com";
            const string articleFileName = @"_posts/2018-05-27-test.md";

            var message = new RegisterCommentNotification(
                commentId,
                userEmail,
                articleFileName);

            // Act
            await this.endpointInstance.Send(message).ConfigureAwait(false);

            // Assert
            Assert.Pass();
        }

        [Test]
        public async Task NotifyAboutCommentAnswer_Send_NoException()
        {
            // Arrange
            const bool isApproved = true;

            var message = new NotifyAboutCommentAnswer(commentId, isApproved);

            // Act
            await this.endpointInstance.Send(message).ConfigureAwait(false);

            // Assert
            Assert.Pass();
        }
    }
}

File: src/tests/integration/Bc.Endpoint.Integration.Tests/CommentAnswerNotificationSender.fs
module Bc.Endpoint.Integration.Tests.CommentAnswerNotification

open System
open System.Threading.Tasks
open Bc.Contracts.Internals.Endpoint.CommentAnswerNotification.Commands
open NServiceBus
open NUnit.Framework

[<Ignore("Only for manual tests")>]
module Sender =

    let commentId = Guid.Parse("9a3a0fdc-3a5a-4862-845c-09792d2b6f9a")

    let sendMessage<'T> (message:'T) =
        async {
            let! endpoint = EndpointFactory.getSenderEndpoint () |> Async.AwaitTask
            do! endpoint.Send(message) |> Async.AwaitTask
            do! endpoint.Stop() |> Async.AwaitTask
        } |> Async.RunSynchronously

    [<Test>]
    let registerCommentNotification_send_noException () =

        // Arrange
        let userEmail = "test@test.com"
        let articleFileName = @"_posts/2018-05-27-test.md"
        let message = RegisterCommentNotification(commentId, userEmail, articleFileName)

        // Act
        sendMessage<RegisterCommentNotification> message

        // Assert
        Assert.Pass()

    [<Test>]
    let notifyAboutCommentAnswer_send_noException () =

        // Arrange
        let isApproved = true
        let message = NotifyAboutCommentAnswer(commentId, isApproved)

        // Act
        sendMessage<NotifyAboutCommentAnswer> message

        // Assert
        Assert.Pass()